src/gcs/gcsBrowserWidget.tsx

```diff
@@ -32,11 +101,46 @@ export class GcsBrowserWidget extends Widget {
         driveName: this.driveName
       }
     );
+
     this.browser.showFileCheckboxes = false;
     this.browser.node.style.height = '100%';
     this.layout = new PanelLayout();
     this.node.style.height = '100%';
     (this.layout as PanelLayout).addWidget(this.browser);
+
+
+    let newFolder = new ToolbarButton({
+      icon: iconGCSNewFolder,
+      className: 'jp-NewFolderIcon',
+      onClick: () => {
+        console.log('Folder New', this.browser.model.path);
+      },
+      tooltip: 'New Folder'
+    });
+
+    // Create a file input element
+    this.fileInput = document.createElement('input');
+    this.fileInput.type = 'file';
+    this.fileInput.multiple = true; // Enable multiple file selection
+    this.fileInput.style.display = 'none';
+
+    // Attach event listener for file selection
+    this.fileInput.addEventListener('change', this.handleFileUpload.bind(this));
+
+    // Append the file input element to the widget's node
+    this.node.appendChild(this.fileInput);
+    let gcsUpload = new ToolbarButton({
+      icon: iconGCSUpload,
+      className: 'jp-UploadIcon',
+      onClick: () => {
```

Comment:
onclick: this.onUploadButtonClick

(this will work once we update to private onUploadButtonClick = () => {...

Reply:
Done